downstairs/src/region.rs

```diff
@@ -378,21 +448,28 @@ impl Extent {
     #[instrument]
     pub fn write(
         &self,
-        offset: Block,
-        data: &[u8],
+        write: &crucible_protocol::Write,
     ) -> Result<(), CrucibleError> {
         let mut inner = self.inner.lock().unwrap();
 
-        self.check_input(offset, data)?;
+        self.check_input(write.offset, &write.data)?;
 
         inner.set_dirty()?;
 
-        let byte_offset = offset.value * self.block_size;
+        let byte_offset = write.offset.value * self.block_size;
 
         inner.file.seek(SeekFrom::Start(byte_offset))?;
-        inner.file.write_all(data)?;
+        inner.file.write_all(&write.data)?;
         inner.file.flush()?;
```

Comment:
I'm now really wondering what this is here for.  Also, should it come after the set_encryption_context below?

Reply:
Yeah, I removed it in 7e263d4. Flushing that file is independent of flushing `set_encryption_context` (which I hope sqlite does unprompted every time!)